### PR TITLE
[CARBONDATA-540]Support insertInto without kettle for spark2

### PR DESCRIPTION
**Pull request by QiangCai**
1. insertInto without kettle for spark2

2. fix test case issue for -Pno-kettle  -Pspark-2.0(and -Pspark-1.5)

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/250/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/257/

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonMetastore.scala [hunk 1]
```diff
@@ -157,7 +157,7 @@ class CarbonMetastore(conf: RuntimeConfig, val storePath: String) {
         CarbonRelation(database, tableIdentifier.table,
           CarbonSparkUtil.createSparkMeta(tables.head.carbonTable), tables.head, alias)
       case None =>
-        LOGGER.audit(s"Table Not Found: ${tableIdentifier.table}")
+        // LOGGER.audit(s"Table Not Found: ${tableIdentifier.table}")
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonMetastore.scala]: remove this

File: processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerColumnar.java [hunk 2]
```diff
@@ -747,11 +753,23 @@ private NodeHolder processDataRowsWithOutKettle(List<Object[]> dataRows)
...
           }
         }
+        BigDecimal value = DataTypeUtil.byteToBigDecimal(b);
+        String[] bigdVals = value.toPlainString().split("\\.");
+        long[] bigDvalue = new long[2];
+        if (bigdVals.length == 2) {
+          bigDvalue[0] = Long.parseLong(bigdVals[0]);
+          BigDecimal bd = new BigDecimal(CarbonCommonConstants.POINT+bigdVals[1]);
+          bigDvalue[1] = (long)(bd.doubleValue()*Math.pow(10, value.scale()));
+          //bigDvalue[1] = Long.parseLong(bigdVals[1]);
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerColumnar.java]: remove this

File: processing/src/main/java/org/apache/carbondata/processing/store/SingleThreadFinalSortFilesMerger.java [hunk 3]
```diff
@@ -134,7 +134,7 @@ public boolean accept(File pathname) {
       }
     });
 
-    if (null == fileList || fileList.length < 0) {
+    if (null == fileList || fileList.length <= 0) {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/store/SingleThreadFinalSortFilesMerger.java]: length will not be less than 0

**Comment by chenliang613** [comment]
@QiangCai  please rebase this PR

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.5.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/348/

**Comment by QiangCai** [comment]
@chenliang613 @jackylk 
Rebase done and fixed comments.

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**